wrapper/src/main/java/software/amazon/jdbc/Driver.java

```diff
@@ -139,13 +158,30 @@ public Connection connect(final String url, final Properties info) throws SQLExc
       PARENT_LOGGER.setLevel(logLevel);
     }
 
-    final TargetDriverDialectManager targetDriverDialectManager = new TargetDriverDialectManager();
-    final TargetDriverDialect targetDriverDialect =
-        targetDriverDialectManager.getDialect(driver, info);
+    TargetDriverDialect targetDriverDialect = configurationProfile == null
+        ? null
+        : configurationProfile.getTargetDriverDialect();
 
-    final ConnectionProvider connectionProvider = new DriverConnectionProvider(driver, targetDriverDialect);
+    if (targetDriverDialect == null) {
```

Comment:
let's add some comment here

// if the dialect defined in the configuration profile is null, we get the dialect back from the properties